alica_engine/include/engine/blackboard/Blackboard.h

```diff
@@ -1,15 +1,180 @@
 #pragma once
 
 #include "BlackboardBlueprint.h"
+#include "engine/Types.h"
+#include "engine/modelmanagement/Strings.h"
 #include <any>
 #include <mutex>
 #include <shared_mutex>
 #include <string>
 #include <type_traits>
 #include <unordered_map>
+#include <variant>
+#include <yaml-cpp/yaml.h>
+
+namespace YAML
+{
+template <>
+struct convert<std::monostate>
+{
+    static Node encode(const std::monostate& rhs)
```

Comment:
can get rid of these methods & instead conditionally enable the compilation of the Parser i.e. using if constexpr